staging/src/k8s.io/kube-aggregator/pkg/controllers/status/available_controller.go

```diff
@@ -276,6 +269,18 @@ func (c *AvailableConditionController) sync(key string) error {
 	}
 	// actually try to hit the discovery endpoint when it isn't local and when we're routing as a service.
 	if apiService.Spec.Service != nil && c.serviceResolver != nil {
+		// TODO create transport from config
+
+		proxyTransport, err := transport.New(c.transportConfig)
+		if err != nil {
+			return err
+		}
+		discoveryClient := &http.Client{
```

Comment:
could we create `discoveryClient` in `NewAvailableConditionController` method?

Reply:
if we do that, won't the cert be non-dynamic through the whole lifetime of available_controller? That's the reason I did it like this. When creating discovery client, one needs to do `transport.New` and I understood that will prevent us from using later-loaded certs

Reply:
> if we do that, won't the cert be non-dynamic through the whole lifetime of available_controller?

hmm, I think that the `discovery client` will be created from a `transportConfig` that has a dynamic function for loading the cert `GetCert`. I think it will be called when a server requests a certificate from a client. Should we test it?